tests/src/providers/testqgspostgresprovider.cpp

```diff
@@ -190,5 +194,112 @@ void TestQgsPostgresProvider::testDecodeDateTimes()
 
 }
 
+void TestQgsPostgresProvider::testQuotedValueBigInt()
+{
+  QgsFields fields;
+  QList<int> pkAttrs;
+  QVariantList vlst;
+  QgsPostgresSharedData *sdata;
+
+  QgsField f0, f1, f2, f3;
+
+  // 4 byte integer
+  f0.setName( "fld_integer" );
+  f0.setType( QVariant::Int );
+  f0.setTypeName( "int" );
+
+  fields.clear();
+  pkAttrs.clear();
+  vlst.clear();
+
+  fields.append( f0 );
+  pkAttrs.append( 0 );
+  vlst.append( 42 );
+
+  sdata = new QgsPostgresSharedData;
```

Comment:
```suggestion
  std::shared_ptr< QgsPostgreSharedData > sdata( new QgsPostgresSharedData() );
```

Reply:
Resolved by commit 07358a0 . Thanks for the hint!